dts/bindings/i2c/st,stm32-i2c-v2.yaml

```diff
@@ -13,3 +13,8 @@ properties:
 
     interrupts:
       required: true
+
+    timing:
+        type: int
+        required: false
+        description: optional a pre-computed I2C_TIMINGR value to override runtime timing calculation
```

Comment:
Maybe a better name would be `timingr`. Would also be nice to include some dts-bindings helpers to, e.g.

```
timingr = <(I2C_STM32_SCLL(x) | I2C_STM32_SCLH(y) | ...)>;
```

Reply:
The way I see things,  users would mostly rely on STM32CubeMX (PC SW tool) which computes the timing value based on peripheral clocks and fall and rise time values. In STM32CubeMX the setting is described as timing so I tried to keep namings aligned.

Reply:
> Based on the use in the next commit I suggest ll-i2c-timing

Can you elaborate on the `ll` prefix? If this relates to the STM32Cube API in use, I'm not sure this a good choice, as this is a pure SW choice while timingr is a register of the V2 I2C IP (I2C_TIMINGR), which needs to be configured whatever the API used (HAL, LL, CMSIS, ...).
Also, IMHO, `-i2c-` feels a bit redundant as for a i2c binding parameter.

Reply:
> I don't like the way this silently overrides any runtime configuration by passing flags like `I2C_SPEED_FAST` to the device that has this flags. This seems likely to be confusing and to cause problems when different bus devices require different speeds.

Right ... the timing option allows very precise and fine-tuned configuration for a complete and definitive HW setup, but it is not dynamic at all. The use case it targets is rather a final product than a development kit.

> A better approach might be to provide an array that specifies a map from a standard speed (100 kHz, 400 kHz, etc) to the desired value to use when programming that selected speed, assuming that the clock configuration is not likely to be runtime controllable.

That would handle a few more cases indeed, but still would not handle a runtime change of the clock tree. 
The default algorithm should rather be used for runtime configuration. 

> I'm also a little concerned with the degree to which v1 and v2 STM I2C drivers diverge in functionality, but maybe that's unavoidable.
v1 and v2 are quite different. This very fine timing capability (together with programming complexity) is V2 only. 

> There is still interest in addressing #4039, which might be a better way to deal with this, though it requires some API redesign.

This timing option was not primarily meant to allow new frequencies, but to allow a product designer to fine-tune the I2C values to precisely match his hardware with measured fall time / rise time . Even with #4039 introduction, this aspects would probably require introduction of a timing option. But then if #4039 is addressed  ... that would become impossible to provide a table of values as suggested earlier ...  

so do you think I should implement the support for 3 values (100/400/1000kHz) as suggested earlier ?

Reply:
> > Based on the use in the next commit I suggest ll-i2c-timing
> 
> Can you elaborate on the `ll` prefix?

The value is passed directly to `LL_I2C_SetTiming()` in the driver.  That underneath that assigns `TIMINGR` in the I2C peripheral register set is true, but nobody will see that unless they look at the HAL implementation.

That there are other APIs than the one used is a good point.  `i2c-` because, again, that's part of the macros and constants used to construct and use the value, but I can see a case for assuming the context is the I2C peripheral structure and eliminating it.  

This isn't a blocking item, so I'll defer to @erwango.  "timingr" still seems a little opaque, but that could be addressed in the property description.